core/model/src/main/java/org/eclipse/rdf4j/model/util/RDFContainers.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+
+package org.eclipse.rdf4j.model.util;
+
+import org.eclipse.rdf4j.model.*;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+import org.eclipse.rdf4j.model.vocabulary.RDF;
+import org.eclipse.rdf4j.model.vocabulary.RDFS;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * TODO add documentation
+ */
+public class RDFContainers {
+
+	public static <C extends Collection<Statement>> C asRDF(IRI containerType, Iterable<?> values, Resource container, C sink,
+                                                            Resource... contexts) {
+
+		Objects.requireNonNull(sink);
+		consumeContainer(containerType, values, container, st -> sink.add(st), contexts);
+		return sink;
+	}
+	public static void consumeContainer(IRI containerType, Iterable<?> values, Resource container, Consumer<Statement> consumer,
+			Resource... contexts) {
+		consumeContainer(containerType, values, container, consumer, SimpleValueFactory.getInstance(), contexts);
+	}
+	public static void consumeContainer(IRI containerType, Iterable<?> values, Resource container, Consumer<Statement> consumer,
+			ValueFactory vf,
+			Resource... contexts) {
+		Objects.requireNonNull(values, "input collection may not be null");
+		Objects.requireNonNull(consumer, "consumer may not be null");
+		Objects.requireNonNull(vf, "injected value factory may not be null");
+
+		Resource current = container != null ? container : vf.createBNode();
+		boolean validType = Objects.equals(containerType, RDF.ALT) ||
+				Objects.equals(containerType, RDF.BAG) ||
+				Objects.equals(containerType, RDF.SEQ);
+
+		assert validType: "containerType should be one of ALT, BAG or SEQ";
+
+		Statements.consume(vf, current, RDF.TYPE, containerType, consumer, contexts);
+
+		Iterator<?> iter = values.iterator();
+		int elementCounter = 1;
+		while (iter.hasNext()) {
+			Object o = iter.next();
+			Value v = o instanceof Value ? (Value) o : Literals.createLiteralOrFail(vf, o);
+			IRI elementCounterPredicate = vf.createIRI(RDF.NAMESPACE, "_" + elementCounter);
+			elementCounter++;
+			Statements.consume(vf, current, elementCounterPredicate, v, consumer, contexts);
+			Statements.consume(vf, (Resource) v, RDFS.MEMBER, current, consumer, contexts);
```

Comment:
Isn't this duplicating information? having `:foo rdf:_1 :bar` already implies that `:foo rdfs:member :bar`. I don't think you need to insert the `rdfs:member` property explicitly. Or do you have a specific use case in mind?

Reply:
Actually we were trying to implement the suggestion in the book Semantic Web Technologies by Pascal Hitzler, Markus Krötzsch and Sebastian Rudolph, page number 61.
The `:container rdf:_1 :element` is for the ordering in `rdf:Seq`.
The `:element rdfs:member :container` is to reduce the time taken to check the membership of an element in a container. If we remove this triple, we would need to iterate over all `:container rdf:_nnn :element` to see if it exists. Otherwise, a single check for `:element rdfs:member :container` will be sufficient.

Reply:
The `:container rdf:_1 :element` is for the ordering in `rdf:Seq`.
The `:element rdfs:member :container` is to reduce the time taken to check the membership of an element in a container. If we remove this triple, we would need to iterate over all `:container rdf:_nnn :element` to see if it exists. Otherwise, a single check for `:element rdfs:member :container` will be sufficient.

Reply:
That makes sense, thanks, and well picked up on that mistake, I completely missed that!